sds_data_manager/stacks/ialirt_processing_stack.py

```diff
@@ -46,78 +48,204 @@ def __init__(
-    def add_compute_resources(self):
-        """EC2 compute environment."""
-        # Define user data script
-        # - Updates the instance
-        # - Installs Docker
-        # - Starts the Docker
-        # - Logs into AWS ECR to pull the image onto the instance
-        # - Pulls the Docker image
-        # - Runs the Docker container
-        user_data = ec2.UserData.for_linux()
-        user_data.add_commands(
-            "yum update -y",
-            "amazon-linux-extras install docker -y",
-            "systemctl start docker",
-            "systemctl enable docker",
-            "$(aws ecr get-login --no-include-email --region us-west-2 | bash)",
-            f"docker pull {self.repo.repository_uri}:latest",
-            f"docker run --rm -d -p 8080:8080 {self.repo.repository_uri}:latest",
+        # Defines the type and port number for the
+        # allowed traffic to the Application Load Balancer
+        # and the EC2 instances.
+        ports = IALIRT_PORTS_TO_ALLOW_CONTAINER_1
+
+        # Add single security group in which
+        # both containers will reside
+        self.ecs_security_group = self.add_ecs_security_group(ports)
+
+        # Add a single security group in which
+        # both application load balancers will reside
+        self.load_balancer_security_group = self.add_load_balancer_security_group(ports)
+
+        # Initialize resources for each container
+        self.add_container_resources("Container1", IALIRT_PORTS_TO_ALLOW_CONTAINER_1)
+
+    def add_container_resources(self, container_name, ports):
+        """Add compute resources and load balancer for a container."""
+        # Add an ecs service and cluster for each container
+        ecs_service, ecs_cluster = self.add_compute_resources(
+            container_name, ports, self.ecs_security_group
+        )
+        # Add load balancer for each container
+        load_balancer = self.add_load_balancer(
+            container_name, ports, ecs_service, self.load_balancer_security_group
+        )
+        # Add autoscaling for each container
+        self.add_autoscaling(container_name, ecs_cluster, load_balancer, ports)
+
+    def add_ecs_security_group(self, ports):
+        """Create and return a security group for containers."""
+        ecs_security_group = ec2.SecurityGroup(
+            self,
+            "IalirtEcsSecurityGroup",
+            vpc=self.vpc,
+            description="Security group for Ialirt",
+            allow_all_outbound=True,
         )
 
-        # Security Group for the EC2 Instance
-        security_group = ec2.SecurityGroup(
+        for port in ports:
+            # Add ingress rule for each port
+            ecs_security_group.add_ingress_rule(
+                peer=ec2.Peer.any_ipv4(),
+                connection=ec2.Port.tcp(port),
+                description="Allow inbound traffic on TCP port",
+            )
+        return ecs_security_group
+
+    def add_load_balancer_security_group(self, ports):
```

Comment:
same here and in other places too. Lot of the functions that had `add` seems to be creating resources.
```suggestion
    def create_load_balancer_security_group(self, ports):
```